extensions/amp-animation/0.1/css-expr-ast.js

```diff
@@ -596,6 +601,111 @@ export class CssTranslateNode extends CssFuncNode {
 
 
 /**
+ * An AMP-specific `width()` and `height()` functions.
+ */
+export class CssDimSizeNode extends CssNode {
+  /**
+   * @param {string} dim
+   * @param {?string=} opt_selector
+   * @param {?string=} opt_selectionMethod Either `undefined` or "closest".
+   */
+  constructor(dim, opt_selector, opt_selectionMethod) {
+    super();
+    /** @const @private */
+    this.dim_ = dim;
+    /** @const @private */
+    this.selector_ = opt_selector || null;
+    /** @const @private */
+    this.selectionMethod_ = opt_selectionMethod || null;
+  }
+
+  /** @override */
+  css() {
+    // No CSS represention.
+    return '';
+  }
+
+  /** @override */
+  isConst() {
+    return false;
+  }
+
+  /** @override */
+  calc(context) {
```

Comment:
All parts of animations (both CSS3 and Web Animations) explicitly only measure things once before starting and do not change in the course of the animation. That's by design. E.g. if you have a keyframe with `transform: translateX(10%)` - it's resolved once, e.g. to `10px` and doesn't change even if `10%` changes. However, `<amp-animation>` itself listens to resize events and restarts the animation. So it's sort of done.